pkg/runtime/wfengine/backend.go

```diff
@@ -0,0 +1,267 @@
+	wfLogger.Warnf("%s: aborting workflow execution", wi.InstanceID)
+
+	// Sending false signals the waiting workflow actor to abort the workflow execution.
+	if channel, ok := wi.Properties[CallbackChannelProperty]; ok {
+		channel.(chan bool) <- false
+	}
+	return nil
+}
+
+// AddNewOrchestrationEvent implements backend.Backend and sends the event e to the workflow actor identified by id.
+func (be *actorBackend) AddNewOrchestrationEvent(ctx context.Context, id api.InstanceID, e *backend.HistoryEvent) error {
+	data, err := backend.MarshalHistoryEvent(e)
+	if err != nil {
+		return err
+	}
+
+	// Send the event to the corresponding workflow actor, which will store it in its event inbox.
+	req := invokev1.
+		NewInvokeMethodRequest(AddWorkflowEventMethod).
+		WithActor(WorkflowActorType, string(id)).
+		WithRawDataBytes(data).
+		WithContentType(invokev1.OctetStreamContentType)
+	defer req.Close()
+
+	resp, err := be.actors.Call(ctx, req)
+	if err != nil {
+		return err
+	}
+	defer resp.Close()
+	return nil
+}
+
+// CompleteActivityWorkItem implements backend.Backend
+func (*actorBackend) CompleteActivityWorkItem(ctx context.Context, wi *backend.ActivityWorkItem) error {
+	// Sending true signals the waiting activity actor to complete the execution normally.
+	wi.Properties[CallbackChannelProperty].(chan bool) <- true
+	return nil
+}
+
+// CompleteOrchestrationWorkItem implements backend.Backend
+func (*actorBackend) CompleteOrchestrationWorkItem(ctx context.Context, wi *backend.OrchestrationWorkItem) error {
+	// Sending true signals the waiting workflow actor to complete the execution normally.
+	wi.Properties[CallbackChannelProperty].(chan bool) <- true
+	return nil
+}
+
+// CreateTaskHub implements backend.Backend
+func (*actorBackend) CreateTaskHub(context.Context) error {
+	return nil
+}
+
+// DeleteTaskHub implements backend.Backend
+func (*actorBackend) DeleteTaskHub(context.Context) error {
+	return errors.New("not supported")
+}
+
+// GetActivityWorkItem implements backend.Backend
+func (be *actorBackend) GetActivityWorkItem(ctx context.Context) (*backend.ActivityWorkItem, error) {
+	// Wait for the workflow actor to signal us with some work to do
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case wi := <-be.activityWorkItemChan:
+		return wi, nil
+	}
+}
+
+// GetOrchestrationRuntimeState implements backend.Backend
+func (*actorBackend) GetOrchestrationRuntimeState(context.Context, *backend.OrchestrationWorkItem) (*backend.OrchestrationRuntimeState, error) {
+	return nil, errors.New("not supported")
+}
+
+// GetOrchestrationWorkItem implements backend.Backend
+func (be *actorBackend) GetOrchestrationWorkItem(ctx context.Context) (*backend.OrchestrationWorkItem, error) {
+	// Wait for the workflow actor to signal us with some work to do
+	select {
+	case <-ctx.Done():
+		return nil, ctx.Err()
+	case wi := <-be.orchestrationWorkItemChan:
```

Comment:
what if the channel got closed, is it ok to process a zero-value orchestrationworkitem?

Reply:
Same question as in my previous response.